docs/configuration/index.md

```diff
@@ -0,0 +1,104 @@
+
+To address the last two issues, Forge fuzzily maps each class, method, field, and parameter to a unique identifier, known as the SRG name, via the [ForgeAutoRenamingTool][fart]. SRG mappings are used in production when the game is being run by the user client.
+
+To allow easier development, ForgeGradle allows the user to choose a mapping set to apply on top of SRG mappings, which we will refer to as human-readable mappings. ForgeGradle knows how to convert the mod jar to SRG mappings for use in production via the `reobf*` task.
+
+The mapping set used can be specified by setting the `mappings` field in the `minecraft` block. The `mappings` field takes in two arguments: `channel` which is the type of the mapping set, and `version` which is the version of the mapping set to apply.
+
+Currently, there are three default mapping sets built into ForgeGradle:
+
+* `official` - This uses the mapping set provided by Mojang. These mappings do not have parameter names and only exist from 1.14 onward.
+* `stable` - This uses a mapping set generated by MCP. These are typically incomplete and no longer exist as of 1.17.
+* `snapshot` - This also is a mapping set generated by MCP, similar to a nightly build of a program. These are also typically incomplete and no longer exist as of 1.17.
+
+!!! note
+    The class names used in production are from `stable` prior to 1.17 and from `official` from 1.17 onwards.
+
+```gradle
+mappings {
+    // Sets the mappings to use those from Mojang for Minecraft 1.19.4.
+    mappings channel: 'official', version: '1.19.4'
+
+    // ...
+}
+```
+
+### Parchment
+
+Parchment is an official project maintained by ParchmentMC which provides open, community-sourced parameter names and javadocs on top of the `official` mapping set. You can learn how setup and use the parchment mapping set on [their website][parchment].
+
+Preparing Run Tasks
+-------------------
+
+Run tasks (`run*`) have two separate pipelines depending on whether they are executed through `gradlew` or a run configuration. By default, there are two tasks that prepare the workspace for execution:
+
+First, there is the `prepare*` tasks which extract and apply the relevant mappings for execution in a development environment. Within an integrated development environment (IDE), the mod sources are not compiled by default, as it is unnecessary. Next, there are the `prepare*Compile` tasks which compiles the mod sources. This is used in conjunction within the `prepare*` tasks to execute a run configuration directly through `gradlew`, or if the relevant setting is enabled through the run configuration.
+
+If your IDE is either Eclipse or IntelliJ IDEA, the run configuration can be configured to execute the `prepare*Compile` tasks before starting the game by setting `enableEclipsePrepareRuns` or `enableIdeaPrepareRuns`, respectively, to `true`.
```

Comment:
```suggestion
If your IDE is either Eclipse or IntelliJ IDEA, the run configuration can be configured to execute the `prepare*` tasks before starting the game by setting `enableEclipsePrepareRuns` or `enableIdeaPrepareRuns`, respectively, to `true`. This will allow you to invoke custom Gradle tasks before your IDE launches the game.
```